packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -1020,9 +1020,23 @@ export default function useAutocomplete(props) {
     }
 
     firstFocus.current = false;
-  };
 
-  const handleInputMouseDown = (event) => {
+    // Check if should open/close combo list upon clear click
+
+    const isCloseIconClicked = event.target?.dataset?.testid === 'CloseIcon';
```

Comment:
Can we just refactor this block to a little bit cleaner?
```js
const shouldClose =
  event.target?.dataset?.testid === 'CloseIcon' ||
  event.target?.querySelectorAll('path').length > 0;

if (shouldClose && open) {
  handlePopupIndicator(event);
} else if (shouldClose && !open) {
  return;
}
...
```

Reply:
I think `const shouldClose = isCloseIconClicked || isPathClicked` is much easier to read!

For the parts below, I wonder if it could be like this:
```
if (shouldClose) {
  if (open) {
    handlePopupIndicator(event);
  }
  return;
}
```

In your snippet, it needs to return in the first `if` as well:
```
if (shouldClose && open) {
  handlePopupIndicator(event);
  return; // <-- It would need to return here
} else if (shouldClose && !open) {
  return;
}

Reply:
I already updated the PR with the snippet sent from @hprobotic, but I agree with your comments @mj12albert, should I switch it back to what it was, or try your suggested solution?

Reply:
@mj12albert I tried having it as your suggestion, but i had to change my code cause there's a lint error if left like that:

![image](https://user-images.githubusercontent.com/23372428/222754863-0ddd2229-f52e-4273-8c71-dbd0e0105a08.png)

Reply:
I pushed a commit and updated it to:
```ts
if (shouldClose) {
    if (open) {
        handlePopupIndicator(event);
        return;
    }
    return;
}
```

neither `shouldClose` nor `open` needs to be repeated, and linter should pass 🙌